### PR TITLE
Bumpup flake

### PR DESCRIPTION
**Pull request by GTrunSec**
ref: Re-structure nix flake

### CODE REVIEW

File: flake.nix [hunk 1]
```diff
@@ -2,42 +2,61 @@
...
+    , nixpkgs
+    , fenix
+    , utils
+    , flake-compat
+    , nix-filter
+    }@inputs:
+      (
+        utils.lib.eachDefaultSystem (
+          system: let
+            fenix = inputs.fenix.packages.${system};
```
Comment (Mic92) [inline flake.nix:24]: GIves us fenix and advantages here over rustc from nixpkgs?

File: flake.nix [hunk 2]
```diff
@@ -2,42 +2,61 @@
   description = "A container debugging tool based on FUSE";
 
   inputs = {
-    naersk.url = "github:nmattia/naersk/master";
-    naersk.inputs.nixpkgs.follows = "nixpkgs";
+    fenix.url = "github:nix-community/fenix";
     nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
     utils.url = "github:numtide/flake-utils";
+    flake-compat.url = "github:edolstra/flake-compat";
+    flake-compat.flake = false;
```
[the remark this answers is not in the document]
Comment (GTrunSec) [reply flake.nix:9]: Just make sure you can use the nix-build

File: flake.nix (hunk 1 above)
Comment (GTrunSec) [reply flake.nix:24]: If you mean here, there's no specific advantage except that you can use the latest version of rustic. Yeh, I can revert it to use rustc from nixpkgs

File: flake.nix (hunk 2 above)
[the remark this answers is not in the document]
Comment (Mic92) [reply flake.nix:8]: but how is relevant for the flake?

[the remark this answers is not in the document]
Comment (GTrunSec) [reply flake.nix:8]: I was using the flake to lock/download the flake-compat src. https://github.com/GTrunSec/cntr/blob/flake/default.nix. 
It looks like you already upgraded the flake. So, I'm going to close this PR.

**Pull request closed without merging**